hugolib/page.go

```diff
@@ -1021,6 +1021,14 @@ func (p *Page) update(f interface{}) error {
 			p.Date = fi.ModTime()
 			p.Params["date"] = p.Date
 		}
+	} else if p.Date.IsZero() && p.s.Cfg.GetBool("useFilenameDateAsFallback") {
```

Comment:
We have a little too many config variables as is, and I think we can drop this one and just check that `filenameDateFallbackPattern` is set. But other than this minor detail, this looks very good.